pkg/lightning/mydump/parquet_parser.go

```diff
@@ -15,568 +15,433 @@
 	"go.uber.org/zap"
 )
 
 const (
-	batchReadRowSize = 32
-
-	// if a parquet if small than this threshold, parquet will load the whole file in a byte slice to
-	// optimize the read performance
-	smallParquetFileThreshold = 256 * 1024 * 1024
-	// jan011970 is the date of unix epoch in julian day,
-	jan011970 = 2440588
-	secPerDay = 24 * 60 * 60
-
-	utcTimeLayout = "2006-01-02 15:04:05.999999Z"
-	timeLayout    = "2006-01-02 15:04:05.999999"
+	// defaultBufSize specifies the default size of skip buffer.
+	// Skip buffer is used when reading data from the cloud. If there is a gap between the current
+	// read position and the last read position, these data is stored in this buffer to avoid
+	// potentially reopening the underlying file when the gap size is less than the buffer size.
+	defaultBufSize = 64 * 1024
 )
 
-// ParquetParser parses a parquet file for import
-// It implements the Parser interface.
-type ParquetParser struct {
-	Reader      *preader.ParquetReader
-	columns     []string
-	columnMetas []*parquet.SchemaElement
-	rows        []any
-	readRows    int64
-	curStart    int64
-	curIndex    int
-	lastRow     Row
-	logger      log.Logger
-
-	readSeekCloser ReadSeekCloser
-}
-
-// readerWrapper is a used for implement `source.ParquetFile`
-type readerWrapper struct {
-	ReadSeekCloser
-	store storage.ExternalStorage
-	ctx   context.Context
-	// current file path
-	path string
+func estimateRowSize(row []types.Datum) int {
+	length := 0
+	for _, v := range row {
+		if v.IsNull() {
+			continue
+		}
+		if v.Kind() == types.KindString {
+			length += len(v.GetBytes())
+		} else {
+			length += 8
+		}
+	}
+	return length
 }
 
-func (*readerWrapper) Write(_ []byte) (n int, err error) {
-	return 0, errors.New("unsupported operation")
+// innerReader defines the interface for reading value with given type T from parquet column reader.
+type innerReader[T parquet.ColumnTypes] interface {
+	ReadBatchInPage(batchSize int64, values []T, defLvls, repLvls []int16) (int64, int, error)
 }
 
-func (r *readerWrapper) Open(name string) (source.ParquetFile, error) {
-	if len(name) == 0 {
-		name = r.path
-	}
-	reader, err := r.store.Open(r.ctx, name, nil)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	return &readerWrapper{
-		ReadSeekCloser: reader,
-		store:          r.store,
-		ctx:            r.ctx,
-		path:           name,
-	}, nil
-}
+type columnDumper interface {
+	SetReader(colReader file.ColumnChunkReader)
 
-func (*readerWrapper) Create(_ string) (source.ParquetFile, error) {
-	return nil, errors.New("unsupported operation")
-}
+	Next(*types.Datum) bool
 
-// bytesReaderWrapper is a wrapper of bytes.Reader used for implement `source.ParquetFile`
-type bytesReaderWrapper struct {
-	*bytes.Reader
-	rawBytes []byte
-	// current file path
-	path string
+	Close() error
 }
 
-func (*bytesReaderWrapper) Close() error {
-	return nil
+type generalColumnDumper[T parquet.ColumnTypes, R innerReader[T]] struct {
```

Comment:
why named as dumper, it seems to be a wrapper around reader, so seems it's still a reader or iterator(it has a `Next`)?